File: app/src/main/java/info/xudshen/android/playground/view/activity/MainActivity.java
package info.xudshen.android.playground.view.activity;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.design.widget.CollapsingToolbarLayout;
import android.support.v7.app.AppCompatActivity;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.support.v7.widget.Toolbar;

import butterknife.BindView;
import butterknife.ButterKnife;
import info.xudshen.android.playground.R;
import info.xudshen.android.playground.view.adapter.SampleItemAdapter;

/**
 * Created by xudong on 2017/1/10.
 */

public class MainActivity extends AppCompatActivity {
    @BindView(R.id.collapsing_toolbar)
    CollapsingToolbarLayout collapsingToolbarLayout;
    @BindView(R.id.toolbar)
    Toolbar toolbar;
    @BindView(R.id.sample_list)
    RecyclerView sampleRecyclerView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        setContentView(R.layout.activity_main);
        ButterKnife.bind(this);

        initViews();
        initEvents();
    }

    private void initViews() {
        setSupportActionBar(toolbar);
//        getSupportActionBar().setDisplayHomeAsUpEnabled(true);
//        getSupportActionBar().setDisplayShowHomeEnabled(true);
        collapsingToolbarLayout.setTitle("Playground");

        sampleRecyclerView.setLayoutManager(new LinearLayoutManager(this));
        sampleRecyclerView.setAdapter(new SampleItemAdapter());
    }

    private void initEvents() {
//        toolbar.setNavigationOnClickListener(view -> onBackPressed());
        ((SampleItemAdapter) sampleRecyclerView.getAdapter()).setOnClickListener(
                (view, viewHolder, position, itemModel) -> {
                    Intent intent = new Intent(MainActivity.this, SampleActivity.class);
                    intent.putExtra(SampleActivity.KEY_INDEX, position);
                    startActivity(intent);
                });
    }

    @Override
    protected void onResume() {
        super.onResume();
    }

    @Override
    protected void onPause() {
        super.onPause();
    }

    @Override
    protected void onDestroy() {
        super.onDestroy();
    }
}

File: app/src/main/java/info/xudshen/android/playground/view/activity/MainActivity.kt
package info.xudshen.android.playground.view.activity

import android.content.Intent
import android.os.Bundle
import android.support.v7.app.AppCompatActivity
import android.support.v7.widget.DefaultItemAnimator
import android.support.v7.widget.LinearLayoutManager
import info.xudshen.android.playground.R
import info.xudshen.android.playground.data.SampleItemDataSource
import info.xudshen.android.playground.recyclerview.adapter2.UUniversalAdapter
import info.xudshen.android.playground.view.adapter.USampleItemModel
import kotlinx.android.synthetic.main.activity_main.*

/**
 * Created by xudong on 2017/1/10.
 */

class MainActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        initViews()
        initEvents()
    }

    private fun initViews() {
        setSupportActionBar(toolbar)
//        supportActionBar?.setDisplayHomeAsUpEnabled(true)
//        supportActionBar?.setDisplayShowHomeEnabled(true)
        collapsing_toolbar.title = "Playground"

        sample_list.layoutManager = LinearLayoutManager(this)
        sample_list.itemAnimator = DefaultItemAnimator()
        sample_list.adapter = UUniversalAdapter()
    }

    private fun initEvents() {
//        toolbar?.setNavigationOnClickListener { onBackPressed() }
        SampleItemDataSource.DATA.map {
            (sample_list.adapter as UUniversalAdapter).addModel(USampleItemModel(it))
        }
        (sample_list.adapter as UUniversalAdapter).setOnItemClickListener { view, pos, model ->
            val intent = Intent(this@MainActivity, SampleActivity::class.java)
            intent.putExtra(SampleActivity.KEY_INDEX, pos)
            startActivity(intent)
        }
    }

    override fun onResume() {
        super.onResume()
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onDestroy() {
        super.onDestroy()
    }
}
